docs/writing_reporters.md

```diff
@@ -77,24 +86,34 @@ end
 
 ### Reacting to events
 
-When consuming events, there are four steps to take into account:
+When consuming events, there are five steps to take into account:
 
-1. Extract event measurements from the event. Measurements are optional, so we must skip reporting that particular measurement if it is not available;
+1. If a filter function has been provided, the reporter MUST record the metric only if the
+   filter function returns `true`.
 
-2. Extract all the relevant tags from the event metadata (if they are supported by the reporter);
+2. Extract event measurements from the event. Measurements are optional, so we must skip
+   reporting that particular measurement if it is not available;
 
-3. Implement the logic specific to the reporter;
+3. Extract all the relevant tags from the event metadata (if they are supported by the reporter);
 
-4. How to react to errors. One option is to let the `handle_event/4` callback fail, but that means we will no longer listen to any future event. Another option is to rescue any error and log them. That's the approach we will take in this example. However, be careful! If an event always contains bad data, then we will log an error every time it is emitted;
+4. Implement the logic specific to the reporter;
+
+5. How to react to errors. One option is to let the `handle_event/4` callback fail, but
+that means we will no longer listen to any future event. Another option is to rescue any
+error and log them. That's the approach we will take in this example. However, be careful!
+If an event always contains bad data, then we will log an error every time it is emitted;
 
 Let's see a simsple handler implementation that takes all of those four items into account:
 
 ```elixir
 def handle_event(_event_name, measurements, metadata, metrics) do
   for metric <- metrics do
     try do
-      if measurement = extract_measurement(metric, measurements) do
-        tags = extract_tags(metric, metadata)
+      measurement = extract_measurement(metric, measurements)
+      tags = extract_tags(metric, metadata)
+
+      if !is_filtered(metric, metadata) do
```

Comment:
I think the semantics here are slightly confusing. “Filter” in Elixir (and in FP) means keeping things in the collection, as in Enum.filter/2. But instead we are rejecting them. We should align those, either by changing the behavior or calling it “reject”.

Reply:
I haven't been comfortable with this name, either. It's describing the wrong thing.

Prometheus has the notion of `keep` and `drop` which take regexes to provide this kind of behavior. Those names feel like they more accurately describe to the user what will happen.

What are your thoughts on these two examples?

```elixir
filter: {:keep | :drop, (:telemetry.event_metadata() -> boolean())} | nil

# or 

keep: (:telemetry.event_metadata() -> boolean()) | nil
drop: (:telemetry.event_metadata() -> boolean()) | nil
```